mirrord/analytics/src/lib.rs

```diff
@@ -66,6 +67,31 @@ impl Analytics {
     }
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AnalyticsHash(#[serde(with = "serde_base64")] Vec<u8>);
+
+impl AnalyticsHash {
+    pub fn from_digest(val: u64) -> Self {
+        AnalyticsHash(val.to_be_bytes().to_vec())
+    }
+
+    pub fn from_base64(val: &str) -> Option<Self> {
```

Comment:
```suggestion
    /// Converts from a base64 `&str`.
    pub fn from_base64(val: &str) -> Option<Self> {
```

But, isn't this a `TryFrom<&str>` implementation? Why as a separate method?